From f85752927a507a9a1858ff9de02b3a9803902b71 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:41 -0400
Subject: [fs] nfsd: Remove nfs4_lock_state(): nfsd4_open and nfsd4_open_confirm

Message-id: <1410898849-22126-158-git-send-email-bfields@redhat.com>
Patchwork-id: 95825
O-Subject: [Patch RHEL7 157/165] nfsd: Remove nfs4_lock_state(): nfsd4_open and nfsd4_open_confirm
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 3234975f477f746c22d076ea178a79ea104b2ca7
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index 5dfa03b..3e6a2e3 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -384,8 +384,6 @@ nfsd4_open(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  if (nfsd4_has_session(cstate))
   copy_clientid(&open->op_clientid, cstate->session);
 
- nfs4_lock_state();
-
  /* check seqid for replay. set nfs4_owner */
  resp = rqstp->rq_resp;
  status = nfsd4_process_open1(&resp->cstate, open, nn);
@@ -470,7 +468,6 @@ out:
  }
  nfsd4_cleanup_open_state(cstate, open, status);
  nfsd4_bump_seqid(cstate, status);
- nfs4_unlock_state();
  return status;
 }
 
diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 733e653..5b3452a 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -4007,9 +4007,6 @@ static void nfsd4_deleg_xgrade_none_ext(struct nfsd4_open *open,
   */
 }
 
-/*
- * called with nfs4_lock_state() held.
- */
 __be32
 nfsd4_process_open2(struct svc_rqst *rqstp, struct svc_fh *current_fh, struct nfsd4_open *open)
 {
@@ -4690,8 +4687,6 @@ nfsd4_open_confirm(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  if (status)
   return status;
 
- nfs4_lock_state();
-
  status = nfs4_preprocess_seqid_op(cstate,
      oc->oc_seqid, &oc->oc_req_stateid,
      NFS4_OPEN_STID, &stp, nn);
@@ -4785,7 +4780,6 @@ put_stateid:
  nfs4_put_stid(&stp->st_stid);
 out:
  nfsd4_bump_seqid(cstate, status);
- nfs4_unlock_state();
  return status;
 }
 
-- 
1.7.1